whois-nrtm4/src/main/java/net/ripe/db/nrtm4/NrtmFileRepo.java

```diff
@@ -0,0 +1,48 @@
+package net.ripe.db.nrtm4;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+
+
+@Service
+public class NrtmFileRepo {
+
+    private final String path;
+
+    NrtmFileRepo(
+        @Value("${nrtm.file.path:/tmp}") final String path
+    ) {
+        this.path = path;
+    }
+
+    boolean checkIfFileExists(final String name) {
+        return new File(path, name).exists();
+    }
+
+    void streamFromFile(final String name, final OutputStream out) throws IOException {
+        final File file = new File(path, name);
+        try (final FileInputStream fis = new FileInputStream(file)) {
+            fis.transferTo(out);
+        }
+    }
+
+    public void storeFile(final String name, final String payload) throws IOException {
+        final File file = new File(path, name);
+        try (final FileOutputStream fos = new FileOutputStream(file)) {
```

Comment:
These looks very generic File functions, we have some in FileHelper class of whois-commons. May be re-use hem or add them there? You can get rid of this extra component?

Reply:
this idea of this class is to encapsulate all the functionality we need for NRTM specific files. FileHelper looks like a generic util class for dealing with Zip files. i suggest it's better to keep this as a dedicated NRTM file manager component rather than putting NRTM-specific functionality into FileHelper, because then it would have two responsibilities -- one common and one NRTM-specific. i think it's better to divide responsibilities than stack them up, especially since the functionality is not 'common'. at least, it shouldn't be common -- the NRTM module should be the only module reading and writing NRTM files.

Reply:
But I think functions in this class has nothing to do with NRTM like getting FileInputStream or even checking if file exists or not, these can easily be converted to static method and used in any where where we need to do something with file stream and we can avoid injecting unncessary beans. Its compnent is just using nrtm path that you can pass from NrtmFileService, thats the NRtm specific component and should reside there.

Reply:
May be create a FileStreamerHelper class ? if you do not like commons, lets put it in nrtm module :)